File: AngularCliCodeFlowPkce/src/app/Models/DealDto.ts
export class DealDto {
  productid: number;
  name: string;
  description: string;
  productname: string;
  oldprice: number;
  newprice: number;
  productlink: string;

  constructor(productid:number, name: string, productname: string, description: string, oldprice: number, newprice: number, productlink: string) {
    this.productid = productid;
    this.name = name;
    this.description = description;
    this.productname = productname;
    this.oldprice = oldprice;
    this.newprice = newprice;
    this.productlink = productlink;
  }

  GetDiscountPercentage(): number {
    return (this.newprice - this.oldprice)*100;
  }
}

File: ResourceServer/Model/Deal/DealDto.cs
namespace ResourceServer.Model.Deal
{
    public class DealDto
    {
        public long ProductId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string ProductName { get; set; } = string.Empty;
        public long OldPrice { get; set; }
        public long NewPrice { get; set; }
        public string ProductLink { get; set; } = string.Empty;
        public string Timestamp { get; set; } = string.Empty;
    }
}
